go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/sigstore-go
 
-go 1.21
+go 1.21.5
```

Comment:
This is interesting.

```shell
~/git/sigstore-go % head -3 go.mod
module github.com/sigstore/sigstore-go

go 1.21
~/git/sigstore-go % go mod tidy
~/git/sigstore-go % head -3 go.mod
module github.com/sigstore/sigstore-go

go 1.21.5
````

So running `go mod tidy` updates it, can't remember ever seeing that before.

Reply:
The reason `go mod tidy` is pinning to 1.21.5 is due to the deps tree. I noticed that even after upgrading to go 1.21.6, it will change it to 1.21.5.

If you run `go mod tidy -go=1.21` (in order to specify the intended go version), it will err:

```
go: github.com/theupdateframework/go-tuf/v2@v2.0.0-20240129093820-4e440e28cdf6 requires go@1.21.5, but 1.21 is requested
```

I would suggest we just roll with go 1.21.5 for the `go` directive for now, as we can't run `go mod tidy` otherwise.

Reply:
I'm a bit new to go, but If you do `go mod vendor` and search for `1.21.5`, it will show that it's `go-tuf-metadata` that explicitly requires `1.21.5`

* https://github.com/rdimitrov/go-tuf-metadata/blob/df718890900a95dccb025e4fb5e28641ee66f108/go.mod#L3

```
## explicit; go 1.21.5
github.com/rdimitrov/go-tuf-metadata/metadata
github.com/rdimitrov/go-tuf-metadata/metadata/config
github.com/rdimitrov/go-tuf-metadata/metadata/fetcher
github.com/rdimitrov/go-tuf-metadata/metadata/trustedmetadata
github.com/rdimitrov/go-tuf-metadata/metadata/updater
```

Reply:
Opened a PR to resolve this (iirc it was motivated by a warning from `govulncheck`)- https://github.com/theupdateframework/go-tuf/pull/588 👍

Reply:
Updated now, go mod is `1.21`.